src/main/java/org/embulk/input/s3/S3FileInputPlugin.java

```diff
@@ -198,7 +198,7 @@ public TransactionalFileInput open(TaskSource taskSource, int taskIndex)
         return new S3FileInput(task, taskIndex);
     }
 
-    private AWSCredentialsProvider getCredentialsProvider(PluginTask task)
```

Comment:
Understood `withForceGlobalBucketAccessEnabled`, but why did you need to change `private` to `protected`?

Reply:
In the current code, we extend this class and re-implement getCredentialsProvider to add some validation checks.

Reply:
I mean in the TD implementation When we use version 0.5.3 `getCredentialsProvider` is protected, we override this function to add some validation and remove some unused Provider base on our use case.

Reply:
* That's TD's internal reason, and this is an OSS repository. Do not bring TD's internal reasons here. The OSS repo does not have reasons to accept TD's internal things.
* That's totally unrelated to "Support S3 Private Link". Do not mix up a different topic in a single PR.

Reply:
If it is really needed, create another PR just for making it `protected`, and give a reasonable description why that is useful even for the OSS repo.

Reply:
Thanks, I revert the change back, let me make another PR for changing it to `protected`.